_posts/2023-11-01-#28368.md

```diff
@@ -7,18 +7,79 @@ authors: [ismaelsadeeq]
+
+* The PR also modifies the [`TransactionAddedToMempool`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/validationinterface.h#L99) event parameter from `CTransactionRef` to a new struct which has the [`CTransactionRef`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/primitives/transaction.h#L421) and other information the fee estimator needs such as the size, fees, and others. There has been an attempt at this PR [#11775](https://github.com/bitcoin/bitcoin/pull/11775)
+
 ## Questions
+
+### Conceptual
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
--->
 
+2. Why is it beneficial to remove `CTxMempool` dependency on `CBlockPolicyEstimator`?
+
+3. Are there any benefits of the `CBlockPolicyEstimator` being a member of `CBlockPolicyEstimator` Are there downsides to removing it?
+
+4. The first attempt [#11775](https://github.com/bitcoin/bitcoin/pull/11775) split the `CValidationInterface` into `CValidationInterface` and `MempoolInterface`. What is the distinction between the two interfaces?
+
+5. Is the split necessary to achieve a background fee estimator? Is it crucial in the long run?
+
+6. Why does this approach omit the split?
+
+7. Can you list the other subscribers to the interface? (Hint: look for classes that inherit from `CValidationInterface`)
+
+8. What does it mean for a class to implement one of the `CValidationInterface` methods? Why is it similar to "subscribing to the event"?
+
+9. What is the difference between implementing a `BlockConnected` and a `NewPoWValidBlock` callback, apart from the fact that they are different events? On which thread is each callback executed?
+
+### Code Review
+
+{:start="10"}
+
+10. In [`4986edb`](https://github.com/bitcoin-core-review-club/bitcoin/commit/4986edb99f8aa73f72e87f3bdc09387c3e516197), Why are we adding a new notification `MempoolTransactionsRemovedForBlock` why can't we use `BlockConnected`?
+
+11. In [`1d116df`](https://github.com/bitcoin-core-review-club/bitcoin/commit/1d116df4c0e021c4c810450e3e5358f34d72940b) a new struct `NewMempoolTransactionInfo` is defined in `kernel/mempool_entry.h` Is that the right place for the struct? What can be a better name for the struct?
+
+12. Are there reasons why the `MempoolTransactionsRemovedForBlock` first parameter is not  `vector<CTxMempoolEntry>`?
+
+13. Are there other alternatives to get the base fee of `CTransactionRef`? Why can't we use it instead of making copies of the base fee in `NewMempoolTransactionInfo`?
```

Comment:
Alternatives to what?

What's "it" int "Why can't we use it"?

Reply:
updated to a clear question